### DISCUSSION TITLE
do i have to deploy static generated pages on node js server when using Fallback : true

### DISCUSSION THREAD

**Discussion by Nouhack** [vercel/next.js#21478] [Help]
hi everyone , im new to nexJS so i started learning it and i like his philosophy but there some thing that i didnt understant:
reading the docs i noticed that they say that you can cache you SSG pages in a CDN and you have to deploy your SSR pages in a NodeJs server thats fine but i realize that not all types of SSG pages you can cached them in a CDN only exported SSG pages can be caches in a CDN  for example:
SSG pages that uses fallback:true to generate that requested  new static page per request  -  and -   when using ISG pages also you have to deploy the page in a Node server .
because  there is an extra work when setting fallback to true or revalidate an SSG page and that extra work must be handled from the node JS server  .
Am I right ?

**Comment by stigkj** [comment]
Yes, you are right. It is only simple SSG pages that can be deployed as static pages without having a running Next.js server.

**Reply by stigkj** [reply]
That is, for incremental static regeneration (ISR) you need the Next.js server. But you still would benefit from having a CDN in front to improve caching, i.e. not having to generate and cache the pages on each instance and on each deployment of the server.
